File: Practice/19/C++/19.cpp
#include <iostream>
#include <string>
using namespace std;

void combine(string letters, string left, short repeats);

short length(0);
string input;
int main()
{
	cin >> length;
	cin >> input;

	combine("", input, length - input.length());
}

void combine(string letters, string left, short repeats)
{
	if (letters.length() != length)
	{
		if (repeats > 0)
		{
			left = input;
		}
		for (char word : left)
		{
			combine(letters + word, string(left).erase(left.find(word), 1), (letters.find(word) != string::npos) ? repeats - 1 : repeats);
		}
	}
	else
	{
		cout << letters << " ";
	}
}

File: Practice/19/Python/19.py
def combine(letters, left, repeats):
    if (len(letters) != length):
        if repeats > 0:
            left = userinput
        for word in left:
            if letters.find(word)!=-1:
                combine(letters + word, left.replace(word,"",1),repeats - 1)
            else:
                combine(letters + word, left.replace(word,"",1),repeats)
    else:
        print(letters,end = " ")

length = int(input())
userinput = input()
combine("",userinput,length - len(userinput))
